### PR TITLE
routing

### PR DESCRIPTION
**Pull request by dariatsvetkova**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

Adding React routing feature for user to navigate between List view and Add item view so that the interface is clearly delinteated between viewing and adding.

## Related Issue

Closes [issue #3](https://github.com/the-collab-lab/tcl-22-smart-shopping-list/issues/3)

## Acceptance Criteria

- [x] `react-router-dom` has been added as a project dependency
- [x] Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- [x] When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
- [x] Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

### After

screencapture-tcl-22-shopping-list-pr18-dt-kk-routing-hbqhe1x3-web-app-list-2021-04-05-16_59_24

## Testing Steps / QA Criteria

To test the feature, use one of the following ways:

1) You can either follow the link to see the preview: https://tcl-22-shopping-list--pr18-dt-kk-routing-hbqhe1x3.web.app
Or launch the app locally from your terminal:
 • Pull down this branch with `git pull origin dt-kk-routing` and check that branch out with `git checkout origin dt-kk-routing`
 • Run `npm install`, then `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

2) At the bottom of the screen, you will see the links to the two views: one for the “list” view, the other for the “add an item” view. As you switch between views, the corresponding link becomes bold, the slug updates to match the view and different h2 elements are displayed. You can also switch between views by manually adding "/list" or "/additem" to the URL.

### CODE REVIEW

**Comment by kkatoch89** [comment]
Our first draft PR!! 🔥

**Review by ablizben — COMMENTED** [review]
Code looks great, guys! I appreciate how you even took the time to add in-line styling to make  the changes more noticeable in the browser. Good job on completing the first PR!

**Comment by laurenmbeatty** [comment]
@dariatsvetkova and @kkatoch89 Looks like there are some merge conflicts to resolve...let us know if you need any help.

**Review by cafloyd — APPROVED** [review]
Looks great! Well done getting the routing working!

**Comment by cafloyd** [comment]
@laurenmbeatty Do you mean there are merge conflicts between this branch and the branch for the second PR this week? Should we typically try to check for merge conflicts between concurrent PRs, or just in whichever PR is approved second?

**Comment by laurenmbeatty** [comment]
> @laurenmbeatty Do you mean there are merge conflicts between this branch and the branch for the second PR this week? Should we typically try to check for merge conflicts between concurrent PRs, or just in whichever PR is approved second?

@cafloyd github will let you know if there are merge conflicts between your branch, and whichever branch you are trying to merge into (`main` in this case). It was a result of your team and theirs naturally needing to work in the same files, so, it just really comes down to whoever ends up with the merge conflicts ends up managing it. If you click on the "resolve conflicts" button (just don't resolve them), you can preview what the merge conflicts are.

image

File: src/App.js
Comment (laurenmbeatty) [inline src/App.js:38]: Love `NavLink`.

File: src/App.js
Comment (laurenmbeatty) [inline src/App.js:40]: They also have an `activeClassName` prop that you can attach styles to via CSS...works the same. 👍

File: src/App.js
Comment (laurenmbeatty) [inline src/App.js:24]: I'm curious about this layout ^ 👀  . What led you to use this CSS, in particular?

File: src/App.js
Comment (laurenmbeatty) [inline src/App.js:26]: Great use of heading hierarchy with the `h1` here and the `h2`s in the components.

**Review by laurenmbeatty — APPROVED** [review]
Excellent work!!!!!! 🎉   Just had a question about the CSS, but looks great!

File: src/App.js
Comment (dariatsvetkova) [reply src/App.js:24]: We initially tried to use flexbox to push navigation to the bottom, but it didn't look good for the items in the middle, hence the complications. I can see now that these styles don't make much sense together and we can achieve the same result by just styling the `nav` with `bottom: 0px; left: 50%;`

Comment (laurenmbeatty) [reply src/App.js:24]: I _think_ you can remove these styles, give the `nav` `width: 100%`, then display flex the `ul`, and `justify-content: space-between` or `space-evenly`...let me verify.

Comment (laurenmbeatty) [reply src/App.js:24]: yes. This is with getting rid of those inline styles and with `space-evenly` (you could also do `padding: 0` on the `ul`. It has a default padding-left of 15px. :
image

**Comment by dariatsvetkova** [comment]
@laurenmbeatty we have resolved the merge conflicts and implemented the styles you suggested. Please take a look.

File: src/App.js
Comment (laurenmbeatty) [inline src/App.js:32]:
```suggestion
            justifyContent: 'space-evenly',
```

Comment (laurenmbeatty) [reply src/App.js:32]: I think this will look slightly nicer : )

File: src/App.js
Comment (kkatoch89) [reply src/App.js]: We totally were overthinking that 😂. Thanks for pointing that out Lauren!

File: src/App.js
Comment (laurenmbeatty) [inline src/App.js:23]: I would check with the other group to see if this works for them, demo-wise to have the input and list on different pages. 

If you are doing this, will you still need the `List` and `AddItem` components?

I might keep the paths more generic like `/list` and `/additem` just in case it's a more generic "shopping" list, rather than groceries-specific. Then maybe you could put your `List` and `AddItem` components back where you had them, and put the `GroceryList` and `AddGroceries` components on one page for now.

**Review by laurenmbeatty — COMMENTED** [review]
Just some things to think about.

File: src/App.js
Comment (kkatoch89) [reply src/App.js:23]: Removing the List/AddItem components plus keeping the generic paths totally makes sense. We'll update to reflect that.
Regarding the input and list being on different pages, we thought one of the acceptance criteria* asked for that; we could have understood it wrong though. I agree with the fact that it doesn't look right though. 

*When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed

File: src/App.js
Comment (kkatoch89) [reply src/App.js:32]: Totally agree!

File: src/App.js
Comment (laurenmbeatty) [reply src/App.js:23]: Cool, yep, I see what you're saying. Good call. Just make sure the other team knows, I think. Nice work! 💯

Comment (dariatsvetkova) [reply src/App.js:23]: My understanding also was that the requirements called for separate views. In [this example](https://www.youtube.com/watch?v=mwj74mE9s64), there are separate views for the list and for adding items. Is this what we are going after? 

I'm also not sure how to implement all the components together (`List`, `AddItem`, `GroceryList` and `AddGroceries`). Should we have `GroceryList` and `AddGroceries` at '/', `List` at '/list' and `AddItem` at '/additem'?

Comment (kkatoch89) [reply src/App.js:23]: @cafloyd, @ablizben & @dariatsvetkova, Lauren brought up a great point regarding the UI of our app so far. You'll notice that because of the criteria I mentioned above, the AddGroceries and AddGroceryList aren't both displaying at the same time. Our React router only displays the corresponding component based on the link that's been clicked on. For apps like this, I've mainly seen this sort of behavior on mobile screen sizes, so it definitely looks a bit off.

Comment (laurenmbeatty) [reply src/App.js:23]: > My understanding also was that the requirements called for separate views. In [this example](https://www.youtube.com/watch?v=mwj74mE9s64), there are separate views for the list and for adding items. Is this what we are going after?
> 
> I'm also not sure how to implement all the components together (`List`, `AddItem`, `GroceryList` and `AddGroceries`). Should we have `GroceryList` and `AddGroceries` at '/', `List` at '/list' and `AddItem` at '/additem'?

@dariatsvetkova Yep, you are right. I was thinking of it as the "connecting to firestore" was a preliminary effort that might be completely re-written in future issues, but the way it's set up you could totally build off of it. I think the way you have it is 100% spot-on.

Comment (dariatsvetkova) [reply src/App.js:23]: @laurenmbeatty got it. I've removed the old components and updated the paths to keep them generic. 

To @kkatoch89's point, I agree that the way the UI is set up is a bit awkward on the desktop. Maybe we should brainstorm a better version as a team and open a new issue to implement it later on?

Comment (laurenmbeatty) [reply src/App.js:23]: Yeah, feel free to create issues on the board. I think this looks amaaaazing and you all rocked it. Very clean, and I'm good with merging :💯 %

Comment (dariatsvetkova) [reply src/App.js:23]: @laurenmbeatty thanks :) Should we get approval from all the mentors before merging?

Comment (laurenmbeatty) [reply src/App.js:23]: Nope, no need!  GO for it @dariatsvetkova

**Pull request merged**
